### PR TITLE
feat(core): Make Part.getId() a prototype method

### PR DESCRIPTION
**Pull request by joostdecock**
As requested by @BenJamesBen in #3282

### CODE REVIEW

File: packages/core/src/part.mjs [hunk 1]
```diff
@@ -78,6 +78,17 @@ Part.prototype.attr = function (name, value, overwrite = false) {
   return this
 }
 
+/**
+ * Gets a free ID to use in the part
+ *
```
Comment (BenJamesBen) [inline packages/core/src/part.mjs:83]: From the code that was previously removed, this line is missing in the comment: 
```
 * @param {string} prefix - An optional prefix to apply to the ID
```

**Review by BenJamesBen — DISMISSED** [review]

File: packages/core/src/part.mjs [hunk 2]
```diff
@@ -78,6 +78,17 @@ Part.prototype.attr = function (name, value, overwrite = false) {
...
+ * @return {string} id - A free ID to use
+ */
+Part.prototype.getId = function (prefix = '') {
+  this.freeId += 1
+
+  return prefix + this.freeId
```
Comment (eriese) [inline packages/core/src/part.mjs:89]: I'm not liking that this is repeated code from
```
/**
 * Returns a closure holding a getId method (returns an ID unused in this part)
 *
 * @return {function} getId - The getId function
 */
Part.prototype.__getIdClosure = function () {
  const self = this
  const method = function (prefix = '') {
    self.freeId += 1

    return prefix + self.freeId
  }

  return method
}
```

One has to reference the other. Your choice which

**Review by eriese — CHANGES_REQUESTED** [review]

File: packages/core/src/part.mjs (hunk 1 above)
Comment (joostdecock) [reply packages/core/src/part.mjs:83]: Added. Good catch

File: packages/core/src/part.mjs (hunk 2 above)
Comment (joostdecock) [reply packages/core/src/part.mjs]: I've called the closure instead.

**Review by BenJamesBen — APPROVED** [review]

**Pull request merged**